azurelinuxagent/ga/exthandlers.py

```diff
@@ -310,11 +311,63 @@ def handle_ext_handlers(self, etag=None):
                 logger.info("Extension handling is on hold")
                 return
 
+        wait_until = datetime.datetime.utcnow() + datetime.timedelta(minutes=DEFAULT_EXT_TIMEOUT_MINUTES)
+        prev_handler = None
+        max_dep_level = max([handler.sort_key() for handler in self.ext_handlers.extHandlers])
+
         self.ext_handlers.extHandlers.sort(key=operator.methodcaller('sort_key'))
         for ext_handler in self.ext_handlers.extHandlers:
-            # TODO: handle install in sequence, enable in parallel
             self.handle_ext_handler(ext_handler, etag)
-    
+
+            # Wait for the extension installation until it is handled.
+            # We wait only for the installation. Not for the uninstallation.
+            # If handled successfully, proceed with the current handler.
+            # Otherwise, skip the rest of the extension installation.
+            dep_level = ext_handler.sort_key()
+            if dep_level >= 0 and dep_level < max_dep_level:
+                if not self.wait_for_handler_successful_completion(ext_handler, wait_until):
+                    logger.warn("Skipping processing extensions")
```

Comment:
Let's add the extension name and rephrase the message to read something similar to "Extension {0} failed or timed out, will skip processing the rest of the extensions"

Reply:
This will be too much logging. We already log the extension name in the wait function. This is to just log that the rest of the extensions will be skipped.

Reply:
Sure, we can leave the extension name out if you prefer so. Let's be explicit about why we are skipping the rest of the extensions, though. Maybe something like "An extension failed or timed out, will skip processing the rest of the extensions". Hopefully this will be more helpful for CSS or a customer reading the log.